components/public-api/typescript/src/metrics.ts

```diff
@@ -403,13 +421,46 @@ export class MetricsReporter {
                 },
                 body: JSON.stringify(data),
                 credentials: "omit",
-            });
-
+            };
+        } catch (e) {
+            this.options.logError("metrics: failed to create request", e);
+            return;
+        }
+        return this.doPost(request);
+    }
+    private async doPost(request: MetricsRequest): Promise<void> {
+        try {
+            const response = await fetch(request.url, request);
             if (!response.ok) {
                 this.options.logError(`metrics: endpoint responded with ${response.status} ${response.statusText}`);
             }
         } catch (e) {
-            this.options.logError("metrics: failed to post", e);
+            this.options.logError("metrics: failed to post, trying again next time", e);
```

Comment:
here is a concern that there will be some kind of another error not necessary network, but we have not see it so far